core/org.wso2.carbon.core/src/main/java/org/wso2/carbon/core/util/SignatureUtil.java

```diff
@@ -164,4 +167,32 @@ private static Certificate getCertificate(String alias) throws Exception {
         return cert;
     }
 
+    /**
+     * This method returns the preferred JCE provider identifier to be used.
+     *
+     * @return jce provider identifier name
+     */
+    private static String getPreferredJceProviderIdentifier() {
+        String provider = System.getProperty(ServerConstants.JCE_PROVIDER_PARAMETER);
+        if (ServerConstants.BOUNCY_CASTLE_FIPS_PROVIDER_IDENTIFIER.equalsIgnoreCase(provider)) {
+            return ServerConstants.BOUNCY_CASTLE_FIPS_PROVIDER_IDENTIFIER;
+        }
+        return ServerConstants.BOUNCY_CASTLE_PROVIDER_IDENTIFIER;
+    }
+
+    /**
+     * This method returns the preferred JCE provider class to be used.
+     *
+     * @return jce provider identifier name
+     */
+    private static String getPreferredJceProviderClass(String providerIdentifier) throws Exception {
+         if (ServerConstants.BOUNCY_CASTLE_FIPS_PROVIDER_IDENTIFIER.equalsIgnoreCase(providerIdentifier)) {
+            return ServerConstants.BOUNCY_CASTLE_FIPS_PROVIDER_CLASS;
+        } else if (ServerConstants.BOUNCY_CASTLE_PROVIDER_IDENTIFIER.equalsIgnoreCase(providerIdentifier)) {
+             return ServerConstants.BOUNCY_CASTLE_PROVIDER_CLASS;
+         } else {
+            throw new Exception("Unsupported JCE provider: " + providerIdentifier);
```

Comment:
shall we use a more specific exception type, such as IllegalArgumentException, to clearly indicate the cause of the error.

Reply:
Added with https://github.com/wso2/carbon-kernel/pull/3550/commits/24668eb3412c1db0392c5428d6c2bddd2822a1c1